src/integrationTest/java/com/mongodb/kafka/connect/MongoSinkConnectorTest.java

```diff
@@ -50,6 +61,74 @@ void testSinkSavesAvroDataToMongoDB() {
         assertProducesMessages(topicName, getCollectionName());
     }
 
+    @Test
```

Comment:
In the following tests we test producing messages and the save order based on the tasks and topic partition configuration.

Kafka maps topic-partitions to tasks, so a single partition topic will be in order, where as a multiple partition topic, with multiple tasks will not be stored in message order.